safecheck.py

```diff
@@ -339,18 +74,31 @@ def md5sum(filename):
     return md5.hexdigest()
 
 
-def check_product_crc(product, manifestfile):
+def s1_check_product_crc(product, manifestfile):
+    """
+    Check CRC part of the product name.
+    Only applies to S1 products
+    """
     expected_crc = format(binascii.crc_hqx(pathlib.Path(manifestfile).read_bytes(), 0xFFFF), '04X')
-    actual_crc = pathlib.Path(product).stem[-4:]
+    # Standard products the CRC is the last 4 chararcters in the productname, before the .SAFE extension
+    # This is as well the 9th subpart of product name while splitting using '_'
+    # On GRD/COG products, an additional '_COG' is added in product name
+    actual_crc = pathlib.Path(product).stem.split('_')[8]
     if expected_crc != actual_crc:
         logger.warning(f"crc in product name '{actual_crc}' does not match crc of manifest file '{expected_crc}'")
         return False
     return True
 
 
-def check_manifest_file(file, schema=None):
+def check_manifest_file(file, schema=None, mission=None):
     if schema is None:
-        schema = builtin_manifest_schema
+        if mission == 'S1':
+            xsd_name = 's1_buildin_manifest.xsd'
+            schema = os.path.join(os.path.dirname(__file__), 'xsd', xsd_name)
+
+        elif mission == 'S2':
+            xsd_name = 's2_buildin_manifest.xsd'
+            schema = '/home/ghajduch/Documents/2023/2023-03/S2/S2-PDGS-TAS-DI-PSD-V14.9_SAFE/resources/xsd/int/esa/safe/sentinel/1.1/xfdu.xsd'
```

Comment:
Local path here:
```diff
-            schema = '/home/ghajduch/Documents/2023/2023-03/S2/S2-PDGS-TAS-DI-PSD-V14.9_SAFE/resources/xsd/int/esa/safe/sentinel/1.1/xfdu.xsd'
+            schema = os.path.join(os.path.dirname(__file__), 'xsd', xsd_name)
```

Reply:
Sorry for this.
I pushed a new version with proper path.